consensus/src/quorum_store/batch_store.rs

```diff
@@ -147,16 +148,71 @@ impl BatchStore {
             persist_subscribers: DashMap::new(),
             expiration_buffer_usecs,
         };
-        let db_content = db_clone
-            .get_all_batches()
-            .expect("failed to read data from db");
+
+        if is_new_epoch {
+            tokio::task::spawn_blocking(move || {
+                Self::gc_previous_epoch_batches_from_db(db_clone, epoch);
+            });
+        } else {
+            Self::gc_expired_batches_from_db(
```

Comment:
Should all gc be `spawn_blocking`? An epoch nearing completion could also have a lot of batches to gc? I see this call eventually calls `spawn_blocking` on `delete_batches` so might as well make things consistent and guard against the get_all_batches being expensive?

Reply:
may be the name is misleading. this call also populates the batch store in memory cache and until we populate this cache we cannot return. so this cannot be spawned async.

Reply:
yeah, it seems more consistent to both spawn_blocking, this branch is for restart right?

Reply:
Ah makes sense. Maybe some renaming/moving code around will make this more clear :) Not a blocker, more a nit

Reply:
oh I didn't see the response lol